idem_azurerm/exec/azurerm/storage/container.py

```diff
@@ -212,6 +324,78 @@ async def create_or_update_immutability_policy(
     return result
 
 
+async def upload_blob(
+    hub,
+    ctx,
+    name,
+    container,
+    account,
+    resource_group,
+    file_path,
+    blob_type="BlockBlob",
+    overwrite=False,
+    **kwargs,
+):
+    """
+    .. versionadded:: 3.0.0
+
+    Creates a new blob from a data source with automatic chunking.
+
+    :param name: The blob with which to interact.
+
+    :param container: The name of the blob container.
+
+    :param account: The name of the storage account.
+
+    :param resource_group: The name of the resource group.
+
+    :param file_path: The path of the file to upload to the specified BlobContainer.
+
+    :param blob_type: The type of the blob. Possible values include: "BlockBlob", "PageBlob" or "AppendBlob".
+        The default value is "BlockBlob".
+
+    :param overwrite: Whether the blob to be uploaded should overwrite the current data. If True, upload_blob will
+        overwrite the existing data. If set to False, the operation will fail with ResourceExistsError. The exception
+        to the above is with Append blob types. If set to False and the data already exists, an error will not be raised
+        and the data will be appended to the existing blob. If set True, then the existing append blob will be deleted,
+        and a new one created. Defaults to False.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        azurerm.storage.container.upload_blob test_name test_container test_account test_group test_path
+
+    """
+    result = {}
+
+    blobconn = await hub.exec.azurerm.storage.container.get_client(
+        ctx,
+        client_type="Blob",
+        account=account,
+        resource_group=resource_group,
+        container=container,
+        blob=name,
+        **kwargs,
+    )
+
+    try:
+        with open(file_path, "rb") as data:
+            container = blobconn.upload_blob(
+                data=data, blob_type=blob_type, overwrite=overwrite, **kwargs
+            )
+        data.close()
```

Comment:
working with files using `with` closes the file handle when you exit the block. calling `close()` explicitly is probably redundant. see https://docs.python.org/3/tutorial/inputoutput.html#reading-and-writing-files